File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        for (int i = 0; i < n; i++) {
            String a = binary(n, arr1[i], arr2[i]);
            StringBuilder sb = new StringBuilder();

            for (int j = 0; j < n; j++) {
                if (a.charAt(j) == '0') {
                    sb.append(" ");
                } else {
                    sb.append("#");
                }
            }

            answer[i] = sb.toString();
        }

        return answer;
    }

    private String binary(int n, int a1, int a2) {
        String str = Integer.toBinaryString(a1 | a2);

        StringBuilder sb = new StringBuilder();

        while (sb.length() + str.length() < n) {
            sb.append('0');
        }
        sb.append(str);

        return sb.toString();
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    for a,b in zip(arr1,arr2):
        tmp = []
        a = bin(a)[2:].zfill(n)
        b = bin(b)[2:].zfill(n)
        
        for i in range(n):
            if a[i] == '1' or b[i] == '1':
                tmp.append('#')
            else:
                tmp.append(' ')
        answer.append(''.join(tmp))
    return answer